include/oneapi/dpl/internal/common_config.h

```diff
@@ -16,12 +16,18 @@
 #ifndef _ONEDPL_COMMON_CONFIG_H
 #define _ONEDPL_COMMON_CONFIG_H
 
-#if __cplusplus < 201703L
+#ifndef _MSVC_LANG
+#    define _ONEDPL_CPLUSPLUS __cplusplus
+#else
+#    define _ONEDPL_CPLUSPLUS _MSVC_LANG
+#endif
```

Comment:
```suggestion
#if defined(_MSVC_LANG) && _MSVC_LANG>__cplusplus
#    define _ONEDPL_CPLUSPLUS _MSVC_LANG
#else
#    define _ONEDPL_CPLUSPLUS __cplusplus
#endif
```

Also, maybe name it _ONEDPL__cplusplus to make it more obvious that this is a customized definition of the standard macro